pkg/cleaner/azure/resourcegroup.go

```diff
@@ -0,0 +1,104 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-02-01/resources"
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	// gracePeriod represents the maximum time the CI resources are allowed to
+	// remain up. CI resources older than gracePeriod will be deleted.
+	gracePeriod = 90 * time.Minute
+)
+
+func (c Cleaner) cleanResourceGroup(ctx context.Context) error {
+	// It would be more efficient here to use a filter like "startswith(name,'ci-') or startswith(name,'e2e')"
+	// but this does not seems to work now, see https://github.com/Azure/azure-sdk-for-go/issues/2480.
+	groupIter, err := c.groupsClient.ListComplete(ctx, "", nil)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	deadLine := time.Now().Add(-gracePeriod).UTC()
+
+	for ; groupIter.NotDone(); groupIter.Next() {
+		group := groupIter.Value()
+
+		c.logger.Log("level", "debug", "message", fmt.Sprintf("checking resource group %q", *group.Name))
```

Comment:
I think we should use `LogCtx` always. We may want to add some logging info at some point.

I would also try to comply with https://github.com/giantswarm/fmt/blob/master/go/log_messages.md wherever possible. You can call it `ensuring deletion of CI resource groups`, then `did not delete` etc.